src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao = new CountryDaoImpl(HibernateUtil.getSessionFactory());
+
     @Override
     public Country add(Country country) {
-        return null;
+        countryDao.add(country);
+        return country;
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).orElse(null);
```

Comment:
Returning `null` when the country is not found is not a good practice. It's better to return an `Optional<Country>` from the service method and let the client decide how to handle the absence of a country. This way, you can avoid potential `NullPointerExceptions`.